futures/tests/io_read_line.rs

```diff
@@ -0,0 +1,22 @@
+use futures::executor::block_on;
+use futures::io::AsyncBufReadExt;
+use std::io::Cursor;
+
+#[test]
+fn read_line() {
```

Comment:
Can you also include a test with a reader which isn't always `Ready` in order to make sure the readiness bits are being plumbed through correctly?

Reply:
Thanks to you pointing it out, I was able to find https://github.com/rust-lang-nursery/futures-rs/pull/1566.

Reply:
In addition, I found some issues with the previous read_line implementation. (They have been fixed.)

However, I'd like to copy and rewrite it instead of calling read_until's implementation, to make it safer.